### PR TITLE
QA tests

### PR DESCRIPTION
**Pull request by sjcorbett**
New QA module runs acceptance tests against a configurable Brooklyn. Including the `run-local` profile causes Brooklyn to be launched on localhost with the most recent build of `brooklyn-tosca-transformer` on the classpath.

### CODE REVIEW

**Comment by sjcorbett** [comment]
Strange, seems a new Brooklyn snapshot containing the changes made to `BrooklynApi` to support this work has not been published.

**Comment by robertgmoss** [comment]
Can you provide documentation for how to run these tests?

**Comment by sjcorbett** [comment]
@robertgmoss Done.

File: docs/qa.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+The following assumes you have installed Vagrant. It has been tested with Vagrant 1.8.1.
+
+Create a machine with the following `Vagrantfile`:
+
+    # -*- mode: ruby -*-
+    # vi: set ft=ruby :
+    VAGRANTFILE_API_VERSION = "2"
+    
+    Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+      config.vm.box = "trusty64"
```
Comment (robertgmoss) [inline docs/qa.md]: use the box "ubuntu/trusty64" instead

**Comment by robertgmoss** [comment]
I'm getting strange test failure from the QA module:

```
testBlueprint (classpath://projects/apache-topology.tosca.yaml)
java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.type.TypeFactory.constructType(Ljava/lang/reflect/Type;Ljava/lang/Class;)Lcom/fasterxml/jackson/databind/JavaType;
    at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector._fullDeserializationType(JaxbAnnotationIntrospector.java:1456)
    at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector._findContentAdapter(JaxbAnnotationIntrospector.java:1410)
    at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector.findDeserializationContentConverter(JaxbAnnotationIntrospector.java:1087)
    at com.fasterxml.jackson.databind.introspect.AnnotationIntrospectorPair.findDeserializationContentConverter(AnnotationIntrospectorPair.java:602)
    at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findConvertingContentDeserializer(StdDeserializer.java:995)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.createContextual(MapDeserializer.java:240)
    at com.fasterxml.jackson.databind.DeserializationContext.handleSecondaryContextualization(DeserializationContext.java:685)
    at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:446)
    at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findDeserializer(StdDeserializer.java:946)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:446)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:296)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
    at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
    at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:477)
    at com.fasterxml.jackson.databind.ObjectReader._prefetchRootDeserializer(ObjectReader.java:1856)
    at com.fasterxml.jackson.databind.ObjectReader.forType(ObjectReader.java:659)
    at com.fasterxml.jackson.databind.ObjectReader.withType(ObjectReader.java:716)
    at org.jboss.resteasy.plugins.providers.jackson.ResteasyJackson2Provider.readFrom(ResteasyJackson2Provider.java:121)
    at org.jboss.resteasy.core.interception.AbstractReaderInterceptorContext.readFrom(AbstractReaderInterceptorContext.java:61)
    at org.jboss.resteasy.core.interception.AbstractReaderInterceptorContext.proceed(AbstractReaderInterceptorContext.java:53)
    at org.jboss.resteasy.plugins.interceptors.encoding.GZIPDecodingInterceptor.aroundReadFrom(GZIPDecodingInterceptor.java:59)
    at org.jboss.resteasy.core.interception.AbstractReaderInterceptorContext.proceed(AbstractReaderInterceptorContext.java:55)
    at org.jboss.resteasy.client.core.BaseClientResponse.readFrom(BaseClientResponse.java:442)
    at org.jboss.resteasy.client.core.BaseClientResponse.getEntity(BaseClientResponse.java:390)
    at org.jboss.resteasy.client.core.BaseClientResponse.getEntity(BaseClientResponse.java:361)
    at org.jboss.resteasy.client.core.BaseClientResponse.getEntity(BaseClientResponse.java:355)
    at org.apache.brooklyn.rest.client.util.http.BuiltResponsePreservingError.copyResponseAndClose(BuiltResponsePreservingError.java:57)
    at org.apache.brooklyn.rest.client.BrooklynApi$3.invoke(BrooklynApi.java:271)
    at com.sun.proxy.$Proxy41.createFromYaml(Unknown Source)
    at org.apache.brooklyn.rest.client.BrooklynApiUtil.deployBlueprint(BrooklynApiUtil.java:52)
    at io.tosca.AlienSamplesAcceptanceTest.testBlueprint(AlienSamplesAcceptanceTest.java:60)
... Removed 28 stack frames
testBlueprint (classpath://projects/php-topology.tosca.yaml)
java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.type.TypeFactory.constructType(Ljava/lang/reflect/Type;Ljava/lang/Class;)Lcom/fasterxml/jackson/databind/JavaType;
    at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector._fullDeserializationType(JaxbAnnotationIntrospector.java:1456)
    at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector._findContentAdapter(JaxbAnnotationIntrospector.java:1410)
    at com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector.findDeserializationContentConverter(JaxbAnnotationIntrospector.java:1087)
    at com.fasterxml.jackson.databind.introspect.AnnotationIntrospectorPair.findDeserializationContentConverter(AnnotationIntrospectorPair.java:602)
    at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findConvertingContentDeserializer(StdDeserializer.java:995)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.createContextual(MapDeserializer.java:240)
    at com.fasterxml.jackson.databind.DeserializationContext.handleSecondaryContextualization(DeserializationContext.java:685)
    at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:446)
    at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findDeserializer(StdDeserializer.java:946)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:446)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:296)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
    at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
    at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:477)
    at com.fasterxml.jackson.databind.ObjectReader._prefetchRootDeserializer(ObjectReader.java:1856)
    at com.fasterxml.jackson.databind.ObjectReader.forType(ObjectReader.java:659)
    at com.fasterxml.jackson.databind.ObjectReader.withType(ObjectReader.java:716)
    at org.jboss.resteasy.plugins.providers.jackson.ResteasyJackson2Provider.readFrom(ResteasyJackson2Provider.java:121)
    at org.jboss.resteasy.core.interception.AbstractReaderInterceptorContext.readFrom(AbstractReaderInterceptorContext.java:61)
    at org.jboss.resteasy.core.interception.AbstractReaderInterceptorContext.proceed(AbstractReaderInterceptorContext.java:53)
    at org.jboss.resteasy.plugins.interceptors.encoding.GZIPDecodingInterceptor.aroundReadFrom(GZIPDecodingInterceptor.java:59)
    at org.jboss.resteasy.core.interception.AbstractReaderInterceptorContext.proceed(AbstractReaderInterceptorContext.java:55)
    at org.jboss.resteasy.client.core.BaseClientResponse.readFrom(BaseClientResponse.java:442)
    at org.jboss.resteasy.client.core.BaseClientResponse.getEntity(BaseClientResponse.java:390)
    at org.jboss.resteasy.client.core.BaseClientResponse.getEntity(BaseClientResponse.java:361)
    at org.jboss.resteasy.client.core.BaseClientResponse.getEntity(BaseClientResponse.java:355)
    at org.apache.brooklyn.rest.client.util.http.BuiltResponsePreservingError.copyResponseAndClose(BuiltResponsePreservingError.java:57)
    at org.apache.brooklyn.rest.client.BrooklynApi$3.invoke(BrooklynApi.java:271)
    at com.sun.proxy.$Proxy41.createFromYaml(Unknown Source)
    at org.apache.brooklyn.rest.client.BrooklynApiUtil.deployBlueprint(BrooklynApiUtil.java:52)
    at io.tosca.AlienSamplesAcceptanceTest.testBlueprint(AlienSamplesAcceptanceTest.java:60)
... Removed 28 stack frames
```

**Comment by sjcorbett** [comment]
This is a new one to me. Are you using Brooklyn master?

**Comment by robertgmoss** [comment]
Yes, pulled yesterday.

**Comment by robertgmoss** [comment]
Also @sjcorbett, it didn't like that I had set a password on my Brooklyn. Can we handle this case?

**Comment by sjcorbett** [comment]
Yes, you need to set `-Dusername=foo -Dpassword=bar` when running the tests.

**Comment by sjcorbett** [comment]
Can you test with the absolute latest version of Brooklyn please?

**Comment by robertgmoss** [comment]
Thanks

**Pull request merged**